### ISSUE TITLE
Tutorial does not consider auto-layout

### ISSUE THREAD

**Issue by sniis84** [i-schuetz/SwiftCharts#30]
Very nice and useful tutorials, I'm trying to learn using your library with them.
However I cannot get how to use auto layout.
Is there a simple way to solve this not programmatically?
Thanks

**Comment by i-schuetz** [comment]
For what exactly do you want to use auto layout?

If it's for the chart's view, you can pass a view to the initializer instead of a frame. The view can come from a storyboard or a xib. You can use auto layout normally.

Note that you have to set the view's class to ChartBaseView (in the identity inspector).

**Comment by sniis84** [comment]
I added UIView in the storyboard setting class as ChartBaseView
@IBOutlet weak var viewArea: ChartBaseView!
Then
let chartFrame = viewArea.bounds
Finally
viewArea.addSubview(chart.view)

But it does not work, which method should I use? Thanks

**Comment by i-schuetz** [comment]
You have to pass viewArea to the chart:

```swift
        let chart = Chart(
            view: viewArea,
            layers: [...]
```

And since viewArea is already in your view hierarchy, you don't have to add it again. `chart.view` is `viewArea`.

**Comment by sniis84** [comment]
I did that but I stop seeing axes and grid.
Possible?

**Comment by i-schuetz** [comment]
You are probably using the view before its frame is calculated. When you use auto layout this doesn't happen immediately. Try putting the chart initialization in `viewDidLayoutSubviews`. You will need a flag to check if it ran already because it's called multiple times. Here is a gist: https://gist.github.com/i-schuetz/f6e285eb0c86bb831510

Alternatively you can use `viewDidAppear`, which is called only once, but this happens a bit too late in the display process and there will be a noticeable delay.

**Comment by sniis84** [comment]
You closed this, but the question has not been solved!

**Comment by i-schuetz** [comment]
Since you stopped answering and opened a new issue where you recently mentioned that you see axis values (which solves your last comment) I assumed this was solved. What is missing?

**Comment by sniis84** [comment]
I solved the axis problem. But the auto layout has not been solved. As I've already replayed to your other link. Your tutorial about autolayout does not work properly. The image is stretched and not proportionally rotated.

**Comment by i-schuetz** [comment]
Which image? Rotation? Can you post a full example / description, maybe also a screenshot so I can get an idea of the problem?

**Comment by sniis84** [comment]
You can find full comments and pictures on the link you provided
https://gist.github.com/i-schuetz/f6e285eb0c86bb831510

Thanks

**Comment by i-schuetz** [comment]
Ok, the code was missing handling of rotation change. You didn't mention this. I updated it. Now it resizes correctly. Note that you have the possibility to use different interval sizes depending of the orientation (in case you would for example show more x values, because there is more screen space available).

**Comment by sniis84** [comment]
Could be interesting to activate scroll on the xAxis only when landscape view is activated. I'll try to figure out on my own, in case I'll open a new question :)

**Comment by sniis84** [comment]
Ciao Friedrich,

Sei stato invitato da Nicholas a partecipare alla Coppa Universitaria di Dropbox. La Coppa Universitaria ti permette di collaborare con i tuoi compagni per guadagnare fino a 25 GB di spazio Dropbox gratuito per due anni!

Per guadagnare spazio devi solo invitare gli amici e utilizzare Dropbox per condividere, collaborare e lavorare.

Partecipa alla Coppa Universitaria![2]

Ricorda che devi essere uno studente iscritto presso un'istituzione scolastica di livello universitario per poter partecipare. Ulteriori informazioni sulla Coppa Universitaria.[3]

Buon lavoro con Dropbox!


 
 Lavoriamo insieme, vinciamo insieme.

____________________________________________________
Se preferisci non ricevere inviti da Dropbox, visita questa pagina[4].
Dropbox, Inc., PO Box 77767, San Francisco, CA 94107

[1]: https://www.dropbox.com/l/UPPuNyCmjN8kfShoUlXqmH/campuscup?text=1
[2]: https://www.dropbox.com/l/i5w3Af0Yz9GMTU6GnVSiMr/campuscup?text=1
[3]: https://www.dropbox.com/l/g9cH5xq4vfS0er1k7IQeZj/help/9131?text=1
[4]: https://www.dropbox.com/l/yjEEcFldIvoyzoc2E8Vmzz?text=1

**Comment by sniis84** [comment]
Ciao Marcello,

Sei stato invitato da Nicholas a partecipare alla Coppa Universitaria di Dropbox. La Coppa Universitaria ti permette di collaborare con i tuoi compagni per guadagnare fino a 25 GB di spazio Dropbox gratuito per due anni!

Per guadagnare spazio devi solo invitare gli amici e utilizzare Dropbox per condividere, collaborare e lavorare.

Partecipa alla Coppa Universitaria![2]

Ricorda che devi essere uno studente iscritto presso un'istituzione scolastica di livello universitario per poter partecipare. Ulteriori informazioni sulla Coppa Universitaria.[3]

Buon lavoro con Dropbox!


 
 Lavoriamo insieme, vinciamo insieme.

____________________________________________________
Se preferisci non ricevere inviti da Dropbox, visita questa pagina[4].
Dropbox, Inc., PO Box 77767, San Francisco, CA 94107

[1]: https://www.dropbox.com/l/GKNO3EQtn6nwvXj4PkexpF/campuscup?text=1
[2]: https://www.dropbox.com/l/z7GZHyvXghOY909miU9pPp/campuscup?text=1
[3]: https://www.dropbox.com/l/iwIAS0Y7MObduw302H7l2i/help/9131?text=1
[4]: https://www.dropbox.com/l/vJxKVd1nXaQKPteIy7ctCx?text=1
